### PR TITLE
not set "request_id" if it is None

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by xiangyan99** [comment]
#9545

File: sdk/core/azure-core/tests/test_request_id_policy.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 request_id_set_values = ("bar", None, "_unset")
 request_id_req_values = ("baz", None, "_unset")
 full_combination = list(product(auto_request_id_values, request_id_init_values, request_id_set_values, request_id_req_values))
+test_combination = (True, 'foo', 'bar', None)
```
Comment (lmazuel) [inline sdk/core/azure-core/tests/test_request_id_policy.py]: What is this?

**Comment by xiangyan99** [comment]
Here is the logic
1. if there is a operation level input, use it (None means ignore req_id_policy)
2.if user sets the request_id in the policy (either in init or call set_req_id), use it (None means ignore. And as long as user sets it, we ignore auto_req_id)
3.if user never sets request_id and auto_req_id is True(True is default value), we generate auto_id
4.if user never sets request_id and set auto_req_id = False, no request id will be sent.

File: sdk/core/azure-core/tests/test_request_id_policy.py (hunk 1 above)
Comment (xiangyan99) [reply sdk/core/azure-core/tests/test_request_id_policy.py]: mistake...

File: sdk/core/azure-core/azure/core/pipeline/policies/_universal.py [hunk 2]
```diff
@@ -149,6 +149,10 @@ def on_request(self, request):
         request_id = unset = object()
         if 'request_id' in request.context.options:
             request_id = request.context.options.pop('request_id')
+            if not request_id:
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/policies/_universal.py]: Note (not feedback):
That will work with `request_id=None` as expected, but also `request_id=False` or `request_id=[]`.
I think it's fair, and that would be overreacting to try to deal with that

**Pull request merged**